conftest.py

```diff
@@ -176,37 +168,61 @@ def function_to_seed_torch(seed):
 
 
 @pytest.fixture(autouse=True)
-def autoseeding_of_everything(record_property, request):
+def autoseeding_of_everything(request):
     """Seed everything we can, for determinism."""
-    main_seed = request.config.getoption("--forcing_random_seed", default=None)
+    randomly_seed = request.config.getoption("--randomly-seed", default=None)
 
-    if main_seed is None:
-        main_seed = random.randint(0, 2**64 - 1)
+    if randomly_seed is None:
+        raise ValueError("why isn't --randomly-seed set?")
 
-    seed = main_seed
-    record_property("main seed", main_seed)
+    # We need to find the relative file path of the test file. It does not look native with request,
+    # so we recompute it.
+    absolute_path = str(request.fspath)
+
+    assert (
+        absolute_path.count("concrete-ml/test") == 1
+    ), f"the absolute path is not as expected {absolute_path=}"
+
+    relative_file_path = absolute_path[
+        absolute_path.find("concrete-ml/test") + len("concrete-ml/") :
+    ]
+
+    # Derive the sub_seed from the randomly_seed and the test name
+    derivation_string = f"{relative_file_path} # {str(request.node.name)} # {randomly_seed}"
```

Comment:
this is the derivation of the subseed: for a test of a function f_i (eg, test_compute_bits_precision) on a configuration c_j (eg, x0-8) of a test file t_k (eg, tests/common/test_utils.py) is computed as some hash(f_i, c_j, t_k, randomly-seed)

Reply:
why not indeed ! is `request.node.name` the actual test + config ? like `tests/common/test_utils.py::test_compute_bits_precision[True-False-5]` ?

Reply:
if we have two files t_k in two different directories with the same filename: it would give the same hash and so subseed. I wanted to keep the "all the tests have different seed" properties, so I included the path as well. But I don't want absolute path, such that, whatever our config (eg, mine or yours), the seeds are the same. In other words, if you clone CML in a dir R and I do it in a dir B, our seeds are the same

Reply:
yes yes I got that, it's just taht your above comment says it computes  " hash(f_i, c_j, t_k, randomly-seed)" but in the code I only see 3 variables

but now I guess  `str(request.node.name)` is actually the `f_i, c_j,` part. `relative_file_path` being `t_k`

Reply:
yes: eg, if you launch `python -m pytest -xsvv "tests/common/test_utils.py::test_compute_bits_precision[x0-8]"`, you'll see

```
Using randomly_seed=2216371280
Using derivation_string='tests/common/test_utils.py # test_compute_bits_precision[x0-8] # 2216371280'
Using sub_seed=16202174809221803539
```

so everything is here: the relative test file, the config, the function and the randomly seed.

Reply:
ok great !